models/solana_utils/solana_utils_daily_balances.sql

```diff
@@ -0,0 +1,35 @@
+ {{
+  config(
+        alias='daily_balances',
+        materialized='incremental',
+        file_format = 'delta',
+        incremental_strategy='merge',
+        unique_key = ['token_mint_address', 'address','day'],
+        post_hook='{{ expose_spells(\'["solana"]\',
+                                    "sector",
+                                    "solana_utils",
+                                    \'["ilemi"]\') }}')
+}}
+
+WITH 
+      updated_balances as (
+            SELECT
+                  address 
+                  , date_trunc('day', block_time) as day
+                  , token_mint_address
+                  , post_balance as sol_balance
+                  , post_token_balance as token_balance
+                  , row_number() OVER (partition by address, date_trunc('day', block_time) order by block_slot desc) as latest_balance
+            FROM {{ source('solana','account_activity') }}
+            WHERE tx_success 
+            {% if is_incremental() %}
+            AND block_time >= date_trunc("day", now() - interval '1 week')
+            {% endif %}
+      )
+
+--is there a way to then only keep the most recent "updated_at" for each address?
+SELECT 
+      *
+      , now() as updated_at 
```

Comment:
to your commented question, this `updated_at` value will change for each unique key assigned above when it comes through. since we read a week back each run, it'll update a rolling 7-day window of the unique keys every hour. so since `day` is part of the unique key, this value will change more than just `address`

Reply:
got it - so it won't cause duplicate issues it'll just update this "updated_at" field. Thanks!